pkg/controller/containerjfr/containerjfr_controller_test.go

```diff
@@ -259,6 +259,61 @@ var _ = Describe("ContainerjfrController", func() {
 				checkDeployment(client, true)
 			})
 		})
+		Context("Container jfr has list of certificate secrets", func() {
+			BeforeEach(func() {
+				objs = []runtime.Object{
+					test.NewContainerJFRWithSecrets(), newFakeSecret("testCert1"), newFakeSecret("testCert2"),
+				}
+			})
+			It("Should add volumeMounts to deployment", func() {
+				reconcileFully(client, controller, false)
+				deployment := &appsv1.Deployment{}
+				err := client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, deployment)
+				Expect(err).ToNot(HaveOccurred())
+
+				testSpecs := newFakeServiceSpecs(false)
+				testTLSConfig := newFakeTLSConfig()
+				testContainer := test.NewContainerJFRWithSecrets()
+				expectedDeployment := resource_definitions.NewDeploymentForCR(testContainer, &testSpecs, &testTLSConfig)
+
+				volumeMounts := deployment.Spec.Template.Spec.Containers[0].VolumeMounts
+				expectedVolumeMounts := expectedDeployment.Spec.Template.Spec.Containers[0].VolumeMounts
+				Expect(volumeMounts).To(Equal(expectedVolumeMounts))
+			})
+		})
+		/*
+			Context("Adding a certificate to the TrustedCertSecrets list", func() {
+				BeforeEach(func() {
+					objs = []runtime.Object{
+						test.NewContainerJFR(),
+					}
+				})
+				JustBeforeEach(func() {
+					reconcileFully(client, controller, false)
+				})
+				It("Should update the corresponding deployment", func() {
+					objs = []runtime.Object{
+						test.NewContainerJFRWithSecrets(), newFakeSecret("testCert1"), newFakeSecret("testCert2"),
+					}
```

Comment:
Modifying `objs` within the It block has no effect because the fake client has already been initialized. You could do this instead:
```suggestion
					err := client.Update(context.Background(), test.NewContainerJFRWithSecrets())
					Expect(err).ToNot(HaveOccurred())
```

This should work because `test.NewContainerJFR` and `test.NewContainerJFRWithSecrets` have the same name.

You might want to move the `newFakeSecret` calls to where you initialize `objs` in the `BeforeEach` block.